test/connect.js

```diff
@@ -417,4 +417,40 @@ QUnit.module('@connect with ObserveObject', () => {
 		assert.equal(getTextFromElement(parentDiv), 'abAcdBeCfghi', 'abAcdBeCfghi');
 
 	});
+
+	QUnit.asyncTest('should only use splice on arrays passed into children not objects', (assert) => {
```

Comment:
This test is hard to grok, and the title of the test seems to be describing an implementation detail of the test, not the result we're hoping for the test.

What is breaking, what should it do? Our test title should be what it should do.

Reply:
Maybe something like, `should preserve non-numeric key properties on observable arrays passed as props` or something